cyences_app_for_splunk/appserver/static/privilege_user_role_editor.js

```diff
@@ -0,0 +1,289 @@
+            selected_user_row_events = [];
+        });
+    });
+
+
+    $(document).on("click", "#bulk_edit_selected", function (e) {
+        e.preventDefault();
+        var user_event_uuids = selected_user_row_events;
+        if (user_event_uuids.length > 0) {
+            console.log("launching cyences_user_event_handlers handler with data", user_event_uuids);
+            $(this).trigger("cyences_user_event_handlers", { action: "user_event_edit", user_event_uuids: user_event_uuids });
+        } else {
+            alert("You must select at least one user event.");
+        }
+    });
+
+    $(document).on("click", "#bulk_edit_all", function (e) {
+        e.preventDefault();
+        var user_event_uuids = all_users;
+        if (user_event_uuids.length > 0) {
+            console.log("launching cyences_user_event_handlers handler with data", user_event_uuids);
+            $(this).trigger("cyences_user_event_handlers", { action: "user_event_edit", user_event_uuids: user_event_uuids });
+        } else {
+            alert("You must select at least one user event.");
+        }
+    });
+
+    $(document).on("click", "#bulk_edit_clear", function (e) {
+        e.preventDefault();
+        $("input:checkbox[name=user_row_selector]").prop('checked', false);
+        selected_user_row_events = [];
+    });
+
+    $(document).on("click", "#bulk_edit_select_all", function (e) {
+        e.preventDefault();
+        $("input:checkbox[name=user_row_selector]").prop('checked', true);
+        selected_user_row_events = all_users;
+    });
+
+
+
+    let CyencesUserEventCheckboxRenderer = TableView.BaseCellRenderer.extend({
+        canRender: function (cell) {
+            // Only use the cell renderer for the specific field
+            return (cell.field === "user_row_selector");
+        },
+        render: function ($td, cell) {
+
+            if (cell.field == "user_row_selector") {
+                $td.addClass('user_row_selector');
+
+                // TODO - change this conodition as it is wrong and always go in else. so if you select box change page and again come to same page then checkbox automatically deselect
+                if (_.contains(selected_user_row_events, cell.value)) {
+                    $td.html('<input type="checkbox" class="user_row_selector" id="user_row_selector" name="user_row_selector" value="' + cell.value + '" checked="checked"></input>');
+                } else {
+                    $td.html('<input type="checkbox" class="user_row_selector" id="user_row_selector" name="user_row_selector" value="' + cell.value + '"></input>');
+                }
+                $td.children("input").on("click", function (e) {
+                    e.stopPropagation();
+                    $td.trigger("cyences_user_event_handlers", { "action": cell.field });
+                });
+            }
+            }
+        });
+
+    let CyencesUserEventCSSClassRenderer = TableView.BaseCellRenderer.extend({
```

Comment:
Why do we need two separate cell renderers?

Reply:
make sense. I'll combine it.